lib/util/hud_code_gen.rb

```diff
@@ -141,4 +142,108 @@ module Types::HmisSchema::Enums::Hud
 
     [map_name, lookup_fn_name]
   end
+
+  def generate_hud_list_json(year, excel_file_path)
+    # Parse the existing JSON data. This will be the base from which we start.
+    json_file_path = "lib/data/#{year}_hud_lists.json"
+    json_data = JSON.parse(File.read(json_file_path))
+
+    # TODO: is there a way to better skip these?
+    ignore_codes = ['List', '*', '(see note)']
+    excel_file = Roo::Excelx.new(excel_file_path)
+
+    # Parse Excel file for list codes
+    excel_data = {}.tap do |codes|
+      # Sheet 0: CSV|DE#|Name|Type|List|Null|Notes|Validate
+      # Pull from this sheet to get the names for each list of codes
+      excel_file.sheet(0).each(code: 'List', name: 'Name') do |row|
+        # Clean data coming in from excel
+        code = clean_json_text(row[:code].to_s)
+        list_name = clean_json_text(row[:name].to_s)
+
+        # Only need rows that have referenced codes - not all data on this sheet utilizes a code list
+        unless code.blank? || ignore_codes.include?(code)
+          codes[code] ||= {}
+          (codes[code][:list_name] ||= []) << list_name
```

Comment:
That's some hairy syntax, maybe?
```suggestion
          codes[code] ||= { list_name: [] }
          codes[code][:list_name] << list_name
```